libglusterfs/src/iobuf.c

```diff
@@ -520,6 +551,19 @@ iobuf_get2(struct iobuf_pool *iobuf_pool, size_t page_size)
         page_size = iobuf_pool->default_page_size;
     }
 
+    /* During smallfile testing we have observed the performance
+       is improved significantly while use standard allocation if
+       page size is less than equal to 128KB, the data is available
+       on the link https://github.com/gluster/glusterfs/issues/2771
+    */
+    if (page_size <= (iobuf_pool->default_page_size)) {
```

Comment:
The default page size might change. I suggest 128 *GF_UNIT_KB instead.
```suggestion
    if (page_size <= 128 * GF_UNIT_KB) {
```

Reply:
done